vspace/src/main/java/edu/asu/diging/vspace/web/staff/ModuleController.java

```diff
@@ -1,29 +1,47 @@
 package edu.asu.diging.vspace.web.staff;
 
+import java.util.HashSet;
+import java.util.List;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
-
 import edu.asu.diging.vspace.core.model.IModule;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.impl.ModuleLink;
 import edu.asu.diging.vspace.core.model.impl.ModuleStatus;
+import edu.asu.diging.vspace.core.services.IModuleLinkManager;
 import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+import edu.asu.diging.vspace.core.services.impl.ModuleManager;
+import edu.asu.diging.vspace.web.exception.ModuleNotFoundException;
+import edu.asu.diging.vspace.web.exception.SpaceNotFoundException;
 
 @Controller
 public class ModuleController {
 
     public static final String STAFF_MODULE_PATH = "/staff/module/";
+
     @Autowired
     private IModuleManager moduleManager;
 
+    @Autowired
+    private IModuleLinkManager moduleLink;
+
     @RequestMapping(STAFF_MODULE_PATH+"{id}")
     public String showModule(@PathVariable String id, Model model) {
         IModule module = moduleManager.getModule(id);
         model.addAttribute("module", module);
         model.addAttribute("slides", moduleManager.getModuleSlides(id));
         model.addAttribute("sequences", moduleManager.getModuleSequences(id));
         model.addAttribute("moduleStatus", module.getModuleStatus());
+        HashSet <ISpace> spaces = moduleLink.findModuleLinksFromModuleId(id);
+        if (!spaces.isEmpty()) {
+            
+            model.addAttribute("spacesList", spaces);
```

Comment:
it doesn't hurt adding an empty list here, does it? is the if statement really necessary?

Reply:
If statement is not necessarily required here, so fixed this.